cli/flags.go

```diff
@@ -433,6 +481,14 @@ func init() {
 			visitFlags(cmd, base.AddToFlagMap)
 		})
 
+		// If any of the security flags have been set, clear the insecure
+		// setting. Note that we do the inverse when the --insecure flag is
+		// set. See insecureValue.Set().
+		if cliContext.SSLCA != "" || cliContext.SSLCAKey != "" ||
+			cliContext.SSLCert != "" || cliContext.SSLCertKey != "" {
+			cliContext.Insecure = false
+		}
+
 		cliContext.Addr = net.JoinHostPort(connHost, connPort)
```

Comment:
listening on `:<port>` is the same thing as localhost, so keeping the empty string is fine. Though if we do print it out, I agree that `localhost` is nicer, but I think we print out the address from the listener, not the flag.